pages/content/amp-dev/search.html

```diff
@@ -0,0 +1,155 @@
+---
+$title: Search
+$order: 1
+description: Search amp.dev.
+$view: /views/overview/custom-content.j2
+---
+{% do doc.styles.addCssFile('css/components/templates/custom-content.css') %}
+
+
+{% do doc.styles.addCssFile('/css/components/molecules/biggy.css') %}
+{% do doc.styles.addCssFile('css/components/organisms/teaser-grid.css') %}
+{% do doc.styles.addCssFile('/css/components/molecules/copy.css') %}
+
+{% do doc.amp_dependencies.add('amp-autocomplete', '0.1') %}
+{% do doc.amp_dependencies.add('amp-form', '0.1') %}
+{% do doc.amp_dependencies.add('amp-bind', '0.1') %}
+{% do doc.amp_dependencies.add('amp-list', '0.1') %}
+{% do doc.amp_dependencies.add('amp-mustache', '0.2', 'template') %}
+
+
+
+<section class="ap--container">
+
+  <div class="ap-m-copy ap-m-copy-center-small ap-m-biggy-background">
+    <h1>Search</h1>
+    <form action-xhr="/search/do" class="search-form" id="searchForm" method="GET" on="submit-success: AMP.setState({'query': tempQuery, 'pageNumber': 1, 'showResults': true, 'search': event.response})"
+          target="_top">
+      <amp-autocomplete filter="substring"
+                        min-characters="0"
+                        on="select:AMP.setState({ 'tempQuery': event.value }),searchForm.submit"
+                        submit-on-enter="false"
+                        src="/search/autosuggest"
+      >
+        <input name="q" on="change:AMP.setState({ 'tempQuery': event.value })"
+               placeholder="What are you looking for?"
+               required>
+      </amp-autocomplete>
+      <input name="page" type="hidden" value="1">
+      <input name="locale" type="hidden" value="{{ doc.locale }}">
+      <button type="submit">Search</button>
+    </form>
+    <div [hidden]="showResults != true" hidden>
+      <p [text]="search.result.totalResults + ' results found'" class="info"></p>
```

Comment:
You can move this into the amp-list if you use `single-item` mode.

Reply:
(if that's supported with infinite scroll)

Reply:
Everything that is in amp-list will be rendered with each page load.
But now I moved it into the form where it makes perfect sense.